exporter/clickhouseexporter/exporter_metrics.go

```diff
@@ -37,12 +37,16 @@ func newMetricsExporter(logger *zap.Logger, cfg *Config) (*metricsExporter, erro
 }
 
 func (e *metricsExporter) start(ctx context.Context, _ component.Host) error {
+	internal.SetLogger(e.logger)
+
+	if !e.cfg.ShouldCreateSchema() {
+		return nil
+	}
+
 	if err := createDatabase(ctx, e.cfg); err != nil {
 		return err
 	}
 
-	internal.SetLogger(e.logger)
```

Comment:
Any reason to drop this?

Reply:
It was moved to the top of the function so that it is called before the DDL statements (conditionally) run.

I also considered extracting all of the DDL to its own function outside of `start` so this separation would be more explicit instead of relying on an early return from `ShouldCreateSchema`. Let me know if I should extract it to a `runDDL` type function. 👍